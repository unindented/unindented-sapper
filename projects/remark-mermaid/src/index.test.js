const stringify = require("rehype-stringify");
const remark2rehype = require("remark-rehype");
const markdown = require("remark-parse");
const unified = require("unified");
const mermaid = require(".");

const process = text =>
  new Promise((resolve, reject) => {
    unified()
      .use(markdown)
      .use(mermaid)
      .use(remark2rehype, { allowDangerousHTML: true })
      .use(stringify)
      .process(text, (err, contents) => {
        if (!err) {
          resolve(contents.contents);
        } else {
          reject(err);
        }
      });
  });

describe("remark-mermaid", () => {
  describe("with a valid mermaid block", () => {
    const text = `
\`\`\` mermaid
graph LR;
A-->B;
\`\`\`
`;

    it("replaces it with a diagram", () => {
      return expect(process(text)).resolves.toContain("svg");
    });
  });

  describe("with an invalid mermaid block", () => {
    const text = `
\`\`\` mermaid
graph;
\`\`\`
`;

    it("replaces it with an empty block", () => {
      return expect(process(text)).resolves.toMatchSnapshot();
    });
  });
});
