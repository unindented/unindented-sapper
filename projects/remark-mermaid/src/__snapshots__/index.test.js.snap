// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark-mermaid with an invalid mermaid block replaces it with an empty block 1`] = `"&#x3C;div class=\\"mermaid\\">&#x3C;/div>"`;
